src/configure/helper/controlProvider.ts

```diff
@@ -0,0 +1,30 @@
+import { QuickPickItem, InputBoxOptions } from 'vscode';
+import { IAzureQuickPickOptions } from 'vscode-azureextensionui';
+import { extensionVariables } from '../model/models';
+import { TelemetryHelper } from './telemetryHelper';
+import { TelemetryKeys } from '../resources/telemetryKeys';
+
+export class ControlProvider {
+    private telemetryHelper: TelemetryHelper;
+
+    public constructor(telemetryHelper) {
+        this.telemetryHelper = telemetryHelper;
+    }
+
+    public async showQuickPick<T extends QuickPickItem>(listItems: T[] | Thenable<T[]>, options: IAzureQuickPickOptions, itemCountTelemetryKey?: string): Promise<T> {
+        try {
+            this.telemetryHelper.setTelemetry(TelemetryKeys.CurrentUserInput, options.placeHolder);
+            return await extensionVariables.ui.showQuickPick(listItems, options);
```

Comment:
should you use return with await ? I think you can return directly as extensionVariables.ui.showQuickPick will return Promise<T>.

Reply:
Oh I read about it and my confusion is gone. return await makes sense here :)